noah/src/test/scala/edu/uci/ics/cloudberry/noah/kafka/TestProducer.scala

```diff
@@ -0,0 +1,74 @@
+package edu.uci.ics.cloudberry.noah.kafka
+
+import com.twitter.hbc.core.Client
+import edu.uci.ics.cloudberry.noah.GeneralProducerKafka
+import edu.uci.ics.cloudberry.noah.feed.Config
+import org.apache.kafka.clients.consumer.{ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
+import org.mockito._
+import org.specs2.mock.Mockito
+import org.specs2.mutable.Specification
+import play.api.libs.ws.ahc.AhcWSClient
+import org.mockito.Mockito._
+import com.twitter.hbc.core.endpoint.Location
+
+class TestProducer extends Specification with Mockito {
+
+  "FileProducer" should {
+    "--  read from a .gz file and send records to kafka" in {
+      val generalProducerKafka = mock[GeneralProducerKafka]
+      val kafkaProducer = mock[KafkaProducer[String, String]]
+
+      val mockConfig = mock[Config]
+      val fileProducer = new FileProducer
+      when (mockConfig.getFilePath).thenReturn("./noah/src/test/resources")
```

Comment:
You might want to access the resource folder directly which suggest by [this post](http://stackoverflow.com/a/38935697/2598198).
Otherwise, the test can only run on top folder.